js/main.js

```diff
@@ -106,3 +144,23 @@ $entriesAnchor.addEventListener('click', () => {
 $entryFormAnchor.addEventListener('click', () => {
   viewSwap('entry-form');
 });
+
+$unorderedList.addEventListener('click', event => {
+  if (event.target.tagName === 'I') {
+    const $currLI = event.target.closest('li');
+    const listId = Number($currLI.dataset.entryId);
+    for (let i = 0; i < data.entries.length; i++) {
+      const retrievedEntry = data.entries[i];
+      if (retrievedEntry.entryId === listId) {
+        data.editing = retrievedEntry;
+        break;
+      }
+    }
+    viewSwap('entry-form');
+    $form.elements[0].value = data.editing.title;
+    $form.elements[1].value = data.editing.url;
+    $form.elements[2].value = data.editing.notes;
+    $image.setAttribute('src', data.editing.url);
+    $entryTitle.textContent = 'Edit Entry';
```

Comment:
I would call `viewSwap` after the rest of this, just so we can gurantee we swap views after all the inputs are filled in. Otherwise we might have a slight second or so where we swap views but the inputs are empty